cli/azd/cmd/deploy.go

```diff
@@ -27,33 +27,44 @@ type deployFlags struct {
 	serviceName  string
 	outputFormat *string // pointer to allow delay-initialization when used in "azd up"
 	global       *internal.GlobalCommandOptions
+	*envFlag
```

Comment:
In all the other places, this is just the value `envFlag`, not the pointer.  Is there a reason this one is different?

Reply:
It is intended, but perhaps there's some simpler solution. The reason is due to it's usage in `upFlags`, where we have a single environment flag variable for `up` defined, and yet need to set the flag for `init`, `provision`, and `deploy` separately. We don't have a place to do this copying, so instead it's done via pointer-indirection, where this points to the single instance in `upFlags` in that casae.

Reply:
Ahh - I understand. Thanks!